account/src/main/java/org/killbill/billing/account/api/DefaultImmutableAccountData.java

```diff
@@ -59,4 +64,15 @@ public Currency getCurrency() {
     public DateTimeZone getTimeZone() {
         return dateTimeZone;
     }
+
+    @Override
+    public UUID getParentAccountId() {
+        return parentAccountId;
```

Comment:
You are correct, this is what we suggested to do first, but then we decided to 'hold off' until we clarify the story around unparenting children (in which case caching becomes more complex). So that may come back in the future depending on what we decide.